### ISSUE TITLE
Include ISO codes for sub-divisions

### ISSUE THREAD

**Issue by jamessampford** [ideal-postcodes/postcodes.io#504]
Would be nice if you could include the ISO code and ISO name when doing a lookup

https://www.iso.org/obp/ui/#iso:code:3166:GB

**Comment by cblanc** [comment]
Hey sorry about the wait, didn't catch this

These are the possible values country can take (`select distinct country from postcodes`) and their ISO code(s)

- `Wales` => `GB`
- `Northern Ireland` => `GB`
- `Scotland` => `GB`
- `Channel Islands` => `JE` or `GG`
- `England`  => `GB`
- `Isle of Man` => `IM`

The Channel Islands can be either Jersey (JE) or Guernsey (GY). These can be separated by the outcode. Jersey begin `JE*` and Guernsey begins `GY*`

**Comment by jamessampford** [comment]
Many thanks for the reply, i don’t think I was spot on with the request, I mean could the requests include the ISO codes for the looked up postcode area, such as GB-ABD for Aberdeenshire. Hopefully you have data link this available in your dataset or could possibly marry it up?

All the ISO codes for GB are included in the link

**Comment by cblanc** [comment]
Hi @jamessampford 

As part of the May 2020 update, I spent some time tracking down mappings for ONSPD localities to ISO codes. Unfortunately I'm not able to find an authoritative dataset that could achieve this. The only way to do this was to assemble a manually curated list, cross referencing the source materials and ONSPD by eye

TLDR: I don't think I'll be able to incorporate the information below into the API datasets proper due to the reasons below. However you may find the manually generated mappings useful. It's small and would allow fast translation between a locality and ISO code.

I've put together two mappings `iso-districts.json` and `iso-counties.json` which you may use to attempt to map a postcode to the ISO code. They're uploaded as gists here: https://gist.github.com/cblanc/3c7bc47fd567ac44b8bd04d005ac756b

In order to map a postcode to an ISO code, check the if the `admin_district` is in `iso-districts.json`. If it's not there it's probably a two tier authority so check if the `admin_county` is in `iso-counties.json`.

It's based on the named source linked from iso.org, [Toponymic guidelines for map editors and other editors: united Kingdom of Great Britain and Northern Ireland, 2009 (updated 2019)](https://www.gov.uk/government/publications/toponymic-guidelines/toponymic-guidelines-for-map-and-other-editors-united-kingdom-of-great-britain-and-northern-ireland#administrative-divisions), it appears that these ISO codes relate to unitary authorities. Loosely speaking, these are captured by the `admin_district` properties of the postcode object.

The picture is a little bit more complicated in England because some localities have a two tier system involving counties `admin_county` (as well as districts). So in some instances, it's not possible to look just at `admin_district`.

While putting together this list I came across another problem. The ISO list is slighly behind ONSPD. Bournemout and Poole unitary authorities were merged in 2019. It's now called Bournemouth, Christchurch & Poole. However the list provided by ISO lists them as two separate authorities. For the timebeing I've mapped both UAs to the [Bournemouth code](https://gist.github.com/cblanc/3c7bc47fd567ac44b8bd04d005ac756b#file-iso-districts-json-L22). You may want to tweak that entry or add more complex behaviour if you wish to handle that differently.

As mentioned I'm holding off incorporating this in postcodes.io, mainly because it may be innaccurate:
- It's manually generated so:
  - Human error possible
  - Naming inconsistencies had to be manually tweaked
- Slight lag in data currency, so elements of those lists may break with every ONSPD release

**Comment by jamessampford** [comment]
@cblanc Many thanks for this, will give a look over!
